Logging/.CSProject/LogHandler/AbstractLogHandler.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+
+namespace Anvil.CSharp.Logging
+{
+    /// <summary>
+    /// A type capable of handling logs received by <see cref="Log"/>.
+    /// </summary>
+    public abstract class AbstractLogHandler
```

Comment:
I think I'm ok with this but a couple thoughts came to mind.

### Forcing base type
Now that we're requiring that handlers derive from a specific type. Are we concerned about potential platforms/contexts that would require a different base type? Say a specific type in order to submit logs to their system

### Redundant options
With a common set of formatting options it's a bit weird that you can select redundant options for a handler.
For example, log level and timestamp are baked into Unity's logging API so they aren't really valid options for that handler.

---

If we did want to go back to `ILogHandler` but consolidate the log formatting logic we could create a `LogFormat` or `LogFormatter` type that most handlers would hold as an instance 🤷

Like I said, I'm fine with the change but I just wanted to express the uneasiness I was feeling in case you guys hadn't thought about these specific angles.

Reply:
> ### Forcing base type
 
I did have to change `FileLogHandler` from an `AbstractAnvilBase` to simply `IDisposable`, but I don't think that's a problem - it doesn't need the extra `DisposeSelf` or `Logger` functionality anyway.

We currently have several log handlers but no instances of this problem, and at worst if it comes up later we either rewrite this again, or you use composition instead of inheritance for that special case.

> ### Redundant options

I don't think flexibility is a bad thing. Maybe you don't like how Unity formats their timestamp, or its position in the log, so you want to customize it. Maybe you use a different IDE/C# tool that already includes caller info in the console, so you remove those in your custom handler.

The alternative would be to tailor the context options to each handler, which seems to defeat the purpose of automatically handling log formatting for handlers out of the box.

> `LogFormatter`

If the base is just an interface, how does it provide and use a `LogFormatter` instance? I guess recent C# versions allow default implementations but I'm not sure if you could achieve the same thing. 🤔 If you've got an idea for how to make it work I'd be happy to implement.

Reply:
ok, all good. Was just my 2 cents. No need for changes

> I did have to change FileLogHandler from an AbstractAnvilBase to simply IDisposable, but I don't think that's a problem - it doesn't need the extra DisposeSelf or Logger functionality anyway.

Saw that. I initially thought of asking you to implement `IAnvilDisposable` but it's really not necessary for this use case. The disposal is very straight forward and controlled.

> > `LogFormatter`
> 
> If the base is just an interface, how does it provide and use a `LogFormatter` instance? I guess recent C# versions allow default implementations but I'm not sure if you could achieve the same thing. 🤔 If you've got an idea for how to make it work I'd be happy to implement.

I was thinking the concrete implementation (`FileLogHandler`, `ConsoleLogHandler`, etc..) would just create an instance of `LogFormatter` rather than pulling it from a default implementation. Maybe the developer would even pass in a pre-configured LogFormatter.